src/index.ts

```diff
@@ -1,3 +1,28 @@
-/**
- * DO THE WORK HERE
- */
\ No newline at end of file
+import { fetch, RequestInit } from "undici";
+import { CONSTANTS, ScorecardResult } from "./utils.js";
+
+export async function result(repository: string, opts) {
+  const data = await getScoreCardData(repository, opts);
+
+  return data;
+}
+
+export async function badge(repository: string) {
+  const data = await getScoreCardData(`${repository}/badge`);
+  return data;
+}
+
+async function getScoreCardData(repo: string, opts?: RequestInit) {
+  const response = await fetch(buildURL(repo).href, opts);
+
+  if (response.ok) {
+    return response;
+  } else {
+    throw new Error(response.statusText);
+  }
+}
+
+export function buildURL(repository: string, plateform?: string) {
+  plateform ? "" : (plateform = CONSTANTS.defaultPlatform);
+  return new URL(`${CONSTANTS.baseUrl}/${plateform}/${repository}`);
+}
```

Comment:
platform parameter is not used atm.

```suggestion
export function buildURL(repository: string) {
  const plateform = CONSTANTS.defaultPlatform;
  return new URL(`${CONSTANTS.baseUrl}/${plateform}/${repository}`);
}
```

Reply:
Salut @tony-go, plateform est censé être un paramètre optionnel, telle que @fraxken m'avait suggéré 
await scorecard.result("NodeSecure/scanner", {
platform: "..." // default to github.com
});. 
Du coup  j'aurai dû mettre undefined a la place  ""?